enhancements/teamsync-with-oidc.md

```diff
@@ -0,0 +1,194 @@
+belongs to, if an entry for the group exists in the `TeamSync` table (i.e., team is syncronized with OIDC group) and if user is not already part of the Quay team, the user gets added to the team.
+2. User's quay teams are fetched. The difference between user's quay teams and user's OIDC groups is calculated. For each team the user belongs to (but not part of OIDC groups) and is part of the `TeamSync` table, the user's team membership for that team is removed.
+
+The flowchart for the Login workflow looks as depicted below:
+![](https://github.com/quay/enhancements/assets/11522230/1fcdabf6-a052-4873-9721-1dda4fc2b81d)
+
+**Endpoints in OIDC**
+
+*Authorization endpoint*
+
+According to Oauth Authorization code flow, the authorization server sends a temporary (authorization) code to the client. This code is exchanged for a token. This is a secure flow for web applications with a backend that can store credentials securely. The authorization code flow is depicted below:
+![](https://github.com/quay/enhancements/assets/11522230/9967a80b-77ec-44c4-a5d6-6a0667a8392c)
+Picture credits: https://cloudentity.com/developers/basics/oauth-grant-types/authorization-code-flow/
+
+```
+  GET /authorize?
+    response_type=code%20id_token
+    &client_id=<client_id>
+    &redirect_uri=<redirect_uri>
+    &scope=openid profile email
+    &state=af0ifjsldkj
+```
+`state`: Opaque value used to maintain state between the request and the callback.
+
+Example response:
+```
+  HTTP/1.1 302 Found
+  Location: https://server.example.com:8020/oidcclient/redirect/client01
+    code=SplxlOBeZQQYbYS6WxSbIA
+    &state=af0ifjsldkj
+```
+
+*Token endpoint*
+
+The token endpoint is initiated by the client to obtain `ID token`, `access token` and `refresh token`. This endpoint accepts an authorization_code issued to the client by the authorization endpoint. When the authorization code is validated, appropriate tokens returned in response.
+
+```
+  POST /token
+    Content-Type: application/x-www-form-urlencoded
+    Authorization: Basic auth
+
+    grant_type=authorization_code&code=<authorization_code>
+    &redirect_uri=<redirect_uri>
+```
+
+Example response:
+```
+  HTTP/1.1 200 OK
+    Content-Type: application/json
+    {
+     "access_token": "SlAV32hkKG",
+     "token_type": "Bearer",
+     "refresh_token": "8xLOxBtZp8",
+     "expires_in": 3600,
+     "id_token": "eyJ ... zcifQ.ewo ... NzAKfQ.ggW8h ... Mzqg"
+    }
+```
+
+*UserInfo endpoint*
+
+The UserInfo endpoint is a protected resource that returns claims about a user and is authenticated through an `access_token`. The response of the request is returned as a JSON object.
+
+```
+  GET /userinfo
+    Accept: application/x-www-form-urlencoded
+    access_token=<access_token>
+```
+
+Example response:
+
+```
+  HTTP/1.1 200 OK
+    Content-Type: application/json
+   {
+    "sub"          : "bob",
+    "groupIds"     : [ "bobsdepartment","administrators" ],
```

Comment:
the claim which has list of group is not part of the standard OIDC claims so different auth servers have different names for it. In this example it is `groupIDs` but keycloak has `groups` other providers like Okta give users an option to change the groups claim name. We should add a param in the config which specifies the groups claim name in the `OIDC_CONFIG`

Reply:
Ah! Thanks, I wasn't aware of this. I added in this commit - https://github.com/quay/enhancements/commit/59b08162ece24e61719c6f2b062d5d2be6011edf